src/devices/cddadevice.cpp

```diff
@@ -36,24 +39,46 @@ CddaDevice::CddaDevice(const QUrl& url, DeviceLister* lister,
           SLOT(SongsLoaded(SongList)));
   connect(this, SIGNAL(SongsDiscovered(SongList)), model_,
           SLOT(SongsDiscovered(SongList)));
+  //   connect(&cd_device_, SIGNAL(DiscChanged()), SLOT(DiscChangeDetected()));
+  cdio_ = cdio_open(url_.path().toLocal8Bit().constData(), DRIVER_DEVICE);
+  Q_ASSERT(cdio_);  // todo: error handling?
```

Comment:
I still don't think that assertions are right for this. The cdio_open documentation states that it can return null on an error. What if, for example, the device open failed with EBUSY?
https://git.savannah.gnu.org/cgit/libcdio.git/tree/lib/driver/_cdio_generic.c#n135

Reply:
Right, forgot to replace that with something proper. Done now.